test/Microsoft.AspNet.Mvc.Core.Test/ModelBinding/Validation/DefaultObjectValidatorTests.cs

```diff
@@ -897,23 +958,20 @@ private static IModelValidatorProvider CreateValidatorProvider()
 
         private static DefaultObjectValidator CreateValidator(Type excludedType)
         {
-            var excludeFilters = new List<IExcludeTypeValidationFilter>();
+            var excludeFilters = new List<ValidationExcludeFilter>();
             if (excludedType != null)
             {
-                var excludeFilter = new Mock<IExcludeTypeValidationFilter>();
-                excludeFilter
-                    .Setup(o => o.IsTypeExcluded(It.IsAny<Type>()))
-                    .Returns<Type>(t => t.IsAssignableFrom(excludedType));
-
-                excludeFilters.Add(excludeFilter.Object);
+                excludeFilters.Add(new ValidationExcludeFilter(excludedType));
             }
 
-            return new DefaultObjectValidator(excludeFilters, TestModelMetadataProvider.CreateDefaultProvider());
+            var provider = TestModelMetadataProvider.CreateDefaultProvider(excludeFilters.ToArray());
+            return new DefaultObjectValidator(provider);
         }
 
-        private static DefaultObjectValidator CreateValidator(params IExcludeTypeValidationFilter[] excludeFilters)
+        private static DefaultObjectValidator CreateValidator(params IMetadataDetailsProvider[] providers)
         {
-            return new DefaultObjectValidator(excludeFilters, TestModelMetadataProvider.CreateDefaultProvider());
+            var provider = TestModelMetadataProvider.CreateDefaultProvider(providers);
+            return new DefaultObjectValidator(provider);
```

Comment:
Some cleanup here to make sure that we're always using the same metadata provider in validation and model binding.